### PR TITLE
Recreate price intent after adding to cart

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Refresh price intent after adding an offer to the cart.

## Justify why they are needed

We used to just refresh the page but this doesn't work when we have a static page.

I exposed a `setupPriceIntent` function from the context to avoid having to repeat the code that creates price intents.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1226** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1226" target="_blank">Graphite</a> 
    * **PR #1227** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1227" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1227?utm_source=stack-comment).

File: apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx [hunk 1]
```diff
@@ -48,11 +50,30 @@ export const PurchaseForm = () => {
         }
 
         if (priceIntent.offers.length > 0) {
+          const handleAddedToCart = async (item: ProductOfferFragment) => {
+            notifyProductAdded({
+              name: productData.displayNameFull,
+              price: formatter.money(item.price),
+            })
+
+            const service = priceIntentServiceInitClientSide(apolloClient)
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx:59]: Is there a better place for this code?  `notifyProductAdded` is purely presentational, but priceIntent manipulation can probably live closer to data management

**Review by alebedev — APPROVED** [review]
Nice!  A question about the right place for priceIntent manipulation inside

File: apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx:59]: Yes no super happy about this either -- but perhaps we can sit together and refactor this code more holistically?

**Comment by robinandeer** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: WNWLnrRr37KtikoMmWoN.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1227)

**Pull request merged**